From 4375fe0bd76f6a05c93c3334f8f2ab7aaa7d29f4 Mon Sep 17 00:00:00 2001
From: William Cohen <wcohen@redhat.com>
Date: Mon, 15 Apr 2013 17:22:16 -0400
Subject: [PATCH] Use rmb() for the missing mb() definition

Commit 59d3d7584b2925bd05b4b5d0f4fe89666eb8494a removed the definition of
mb().  mb() was defined as rmb().  This just corrects it back.

Signed-off-by: William Cohen <wcohen@redhat.com>
---
 src/linux-lock.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/linux-lock.h b/src/linux-lock.h
index 741e352..6fe1da5 100644
--- a/src/linux-lock.h
+++ b/src/linux-lock.h
@@ -186,8 +186,8 @@ static inline int __arm_papi_spin_lock (volatile unsigned int *lock)
 
   return 0;
 }
-#define _papi_hwd_lock(lck)   { mb(); __arm_papi_spin_lock(&_papi_hwd_lock_data[lck]); mb(); }
-#define _papi_hwd_unlock(lck) { mb(); _papi_hwd_lock_data[lck] = 0; mb(); }
+#define _papi_hwd_lock(lck)   { rmb(); __arm_papi_spin_lock(&_papi_hwd_lock_data[lck]); rmb(); }
+#define _papi_hwd_unlock(lck) { rmb(); _papi_hwd_lock_data[lck] = 0; rmb(); }
 
 #elif defined(__mips__)
 static inline void __raw_spin_lock(volatile unsigned int *lock)
-- 
1.8.1.4

